### PR TITLE
ARM64: only patch exits if within apropriate ranges. always patch stub too

### PR DESCRIPTION
**Pull request by javierguerragiraldez**
Conditional branches have much smaller ranges than unconditional ones.  If a
side trace is beyond range, leave the exit unmodified; for these cases the
exit stub (an unconditional branch) is patched too.

Modelled after similar procedure in PPC version of the same function.

### CODE REVIEW

File: src/lj_asm_arm64.h [hunk 1]
```diff
@@ -1975,47 +1975,62 @@ static void asm_mcode_fixup(MCode *mcode, MSize size)
...
   }
-  lua_assert(cstart != NULL);
-  lj_mcode_sync(cstart, cend);
+  {
+    ptrdiff_t delta = target - px;
+    lua_assert(valid_S26(delta));
+    *px = A64I_B | ((uint32_t)delta & 0x03ffffffu);
+    if (!cstart) cstart = px;
+  }
+  lj_mcode_sync(cstart, px + 1);
```
Comment (sdmaclea) [inline src/lj_asm_arm64.h]: This icache flush is wrong, because cstrart is not set.

Should be `lj_mcode_sync(cstart ? cstart : px, px + 1);`

Comment (javierguerragiraldez) [reply src/lj_asm_arm64.h]: it's set on line 2031

Comment (sdmaclea) [reply src/lj_asm_arm64.h]: Sorry missed that.  I was looking at why the assert was removed.  I jumped to the conclusion that the invariant condition did not hold.

**Comment by MikePall** [comment]
Applied. Thanks!

**Pull request closed without merging**